media.go

```diff
@@ -949,46 +949,40 @@ func (insta *Instagram) UploadPhoto(photo io.Reader, photoCaption string, qualit
 
 func (insta *Instagram) postPhoto(photo io.Reader, photoCaption string, quality int, filterType int, isSidecar bool) (map[string]interface{}, error) {
 	uploadID := time.Now().Unix()
-	photoName := fmt.Sprintf("pending_media_%d.jpg", uploadID)
-	var b bytes.Buffer
-	w := multipart.NewWriter(&b)
-	w.WriteField("upload_id", strconv.FormatInt(uploadID, 10))
-	w.WriteField("_uuid", insta.uuid)
-	w.WriteField("_csrftoken", insta.token)
-	var compression = map[string]interface{}{
-		"lib_name":    "jt",
-		"lib_version": "1.3.0",
-		"quality":     quality,
-	}
-	cBytes, _ := json.Marshal(compression)
-	w.WriteField("image_compression", toString(cBytes))
-	if isSidecar {
-		w.WriteField("is_sidecar", toString(1))
-	}
-	fw, err := w.CreateFormFile("photo", photoName)
+	rndNumber := rand.Intn(9999999999-1000000000) + 1000000000
+	name := strconv.FormatInt(uploadID, 10) + "_0_" + strconv.Itoa(rndNumber)
+	buf := new(bytes.Buffer)
```

Comment:
nitpick:
var buf  bytes.Buffer

Reply:
It seems buf need to be initialize to be use as a reader
`./media.go:960:71: cannot use buf (type bytes.Buffer) as type io.Reader in argument to http.NewRequest`